tests/performance_calculation/test_performance_calculator.py

```diff
@@ -480,3 +482,63 @@ def test_binary_classification_calculate_without_prediction_column():
         _ = calc.calculate(analysis.merge(analysis_targets, on='id'))
     except Exception as exc:
         pytest.fail(f"an unexpected exception occurred: {exc}")
+
+
+def test_oo_behavior_bc_perf_calc(data):  # noqa: D103
+    reference, monitored, targets = data
+    reference2 = reference.copy(deep=True)
+    monitored = monitored.merge(targets, left_index=True, right_index=True)
+    monitored2 = monitored.copy(deep=True)
+    calc = PerformanceCalculator(
+        timestamp_column_name='timestamp',
+        y_pred='y_pred',
+        y_pred_proba='y_pred_proba',
+        y_true='work_home_actual',
+        metrics=['roc_auc', 'f1'],
+        problem_type='classification_binary',
+    )
+    calc.fit(reference2)
+    results = calc.calculate(monitored2)  # noqa: F841
+    pd.testing.assert_frame_equal(monitored, monitored2)
+    pd.testing.assert_frame_equal(reference, reference2)
+
+
+def test_oo_behavior_mc_perf_calc(data):  # noqa: D103
```

Comment:
Name could be a bit more descriptive, as this is a weird thing to test.

Maybe something like test_input_dataframes_are_not_altered_by_calculator